Cds 67816 pipeline rollback (#1867)
harness/developer-hub: docs/platform/8_Pipelines/define-a-failure-strategy-on-stages-and-steps.md

* Step
* Step Group
* Stage

> * Pipeline

Reviewer: Remove Pipeline from here

* **Timeout** and **Post timeout action:** these are available if you selected **Manual Intervention** in Action. Enter the timeout for the failure strategy and the subsequent action to perform.
* **Retry Count** and **Retry Intervals:** these are available if you selected **Retry** in Action. Enter the number of times to retry the step, and the retries intervals.

### Rollback pipelines

> :::note Info

Author: ```suggestion
:::info
```

Currently, this feature is behind the feature flag, `PIPELINE_ROLLBACK`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.

:::

> The Rollback Pipeline failure strategy applies to all stages in a pipeline. 

Author: ```suggestion
The Rollback Pipeline failure strategy applies to all steps and stages in a pipeline. 
```

> ::: info

Author: ```suggestion
:::info
```

> During [Pipeline chaining](/docs/platform/pipelines/pipeline-chaining/), if there is a stage of type, pipeline, the deployment stages inside the child pipeline cannot be rolled back using the Rollback Pipeline failure strategy because the execution of the child pipeline is considered as a seperate execution.

Reviewer 2: See above

* **Timeout** and **Post timeout action:** these are available if you selected **Manual Intervention** in Action. Enter the timeout for the failure strategy and the subsequent action to perform.
* **Retry Count** and **Retry Intervals:** these are available if you selected **Retry** in Action. Enter the number of times to retry the step, and the retries intervals.

### Add a pipeline failure strategy

The pipeline failure strategy applies to all stages in the pipeline. 

> 

Reviewer: Can we add the feature flag for pipeline rollback over here. It is PIPELINE_ROLLBACK

In a stage in a pipeline, click **Advanced**.

In **Failure Strategy**, you can see the default stage strategy:

**On all errors other than those specified in failure strategies defined here, perform action**

To add an additional stage failure strategy, click **Add**.

In **On failure of type**, select one or more of the error types, or select **All Errors**. Go to [Error types](/docs/platform/Pipelines/w_pipeline-steps-reference/step-failure-strategy-settings#error-types) for more information.

In **Perform Action**, select **Rollback Pipeline**

> 

Reviewer: We need to add a note that this is currently only supported for a few swimlanes: Kubernetes, Native Helm, and ECS.
Also, we need to mention that this only works on Deployment Stages. If there is a stage of type Pipeline (referring to Pipeline Chaining here), then the deployment stages inside the child Pipeline will not be rolled back as it is a separate and completed execution

Reviewer: The list of supported swimlanes will increase to all swimlanes in the next release so we can update the documentation then

* **Timeout** and **Post timeout action:** these are available if you selected **Manual Intervention** in Action. Enter the timeout for the failure strategy and the subsequent action to perform.
* **Retry Count** and **Retry Intervals:** these are available if you selected **Retry** in Action. Enter the number of times to retry the step, and the retries intervals.

### Rollback pipelines

:::note Info

Currently, this feature is behind the feature flag, `PIPELINE_ROLLBACK`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.

:::

The Rollback Pipeline failure strategy applies to all stages in a pipeline. 

::: info

During [Pipeline chaining](/docs/platform/pipelines/pipeline-chaining/), if there is a stage of type, pipeline, the deployment stages inside the child pipeline cannot be rolled back using the Rollback Pipeline failure strategy because the execution of the child pipeline is considered as a seperate execution.

:::

> The Rollback Pipeline failure strategy applies to the following deployments only:

Reviewer: Can we say that this is "As of now", just to indicate the support for other swimlanes is coming soon

> The Rollback Pipeline failure strategy applies to the following deployments only:

Author: ```suggestion
Currently, the Rollback Pipeline failure strategy applies to the following deployments only:
```

* Kubernetes
* Native Helm
* Amazon Elastic Container Service (ECS)

> In a stage in a pipeline, select **Advanced**.

Reviewer: In a stage or step*

> In a stage in a pipeline, select **Advanced**.

Author: ```suggestion
In a pipeline's step or stage, select **Advanced**.
```
